queries.sql

```diff
@@ -228,3 +228,144 @@ GROUP BY
 ORDER BY
     animal_count DESC
 LIMIT 1;
+
+
+--Write queries to answer the following:
+-- Who was the last animal seen by William Tatcher?
+SELECT
+    a.name AS last_animal_seen
+FROM
+    visits v
+JOIN
+    animals a ON v.animal_id = a.id
+WHERE
+    v.vet_id = (SELECT id FROM vets WHERE name = 'William Tatcher')
+ORDER BY
+    v.visit_date DESC
+LIMIT 1;
+
+
+-- How many different animals did Stephanie Mendez see?
+SELECT
+    COUNT(DISTINCT v.animal_id) AS number_of_animals_seen
+FROM
+    visits v
+WHERE
+    v.vet_id = (SELECT id FROM vets WHERE name = 'Stephanie Mendez');
+
+    
+-- List all vets and their specialties, including vets with no specialties.
+SELECT
+    v.name AS vet_name,
+    COALESCE(s.name, 'No Specialty') AS specialty_name
+FROM
+    vets v
+LEFT JOIN
+    specializations sp ON v.id = sp.vet_id
+LEFT JOIN
+    species s ON sp.species_id = s.id
+ORDER BY
+    v.name, specialty_name;
+
+
+-- List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020.
+SELECT
+    a.name AS animal_name
+FROM
+    visits v
+JOIN
+    animals a ON v.animal_id = a.id
+WHERE
+    v.vet_id = (SELECT id FROM vets WHERE name = 'Stephanie Mendez')
+    AND v.visit_date BETWEEN '2020-04-01' AND '2020-08-30';
+
+
+
+-- What animal has the most visits to vets?
+SELECT
+    a.name AS animal_name,
+    COUNT(v.animal_id) AS visit_count
+FROM
+    visits v
+JOIN
+    animals a ON v.animal_id = a.id
+GROUP BY
+    a.name
+ORDER BY
+    visit_count DESC
+LIMIT 1;
+
+
+-- Who was Maisy Smith's first visit?
+SELECT
+    a.name AS first_visited_animal
+FROM
+    visits v
+JOIN
+    animals a ON v.animal_id = a.id
+WHERE
+    v.vet_id = (SELECT id FROM vets WHERE name = 'Maisy Smith')
+ORDER BY
+    v.visit_date
+LIMIT 1;
+
+
+-- Details for most recent visit: animal information, vet information, and date of visit.
+SELECT
+    a.name AS animal_name,
+    v.visit_date,
+    ve.name AS vet_name
+FROM
+    visits v
+JOIN
+    animals a ON v.animal_id = a.id
+JOIN
+    vets ve ON v.vet_id = ve.id
+ORDER BY
+    v.visit_date DESC
+LIMIT 1;
+
+
+-- How many visits were with a vet that did not specialize in that animal's species?
+SELECT
+    COUNT(*) AS num_visits_without_specialization
+FROM
+    visits v
+JOIN
+    animals a ON v.animal_id = a.id
+JOIN
+    vets ve ON v.vet_id = ve.id
+LEFT JOIN
+    specializations s ON ve.id = s.vet_id AND a.species_id = s.species_id
+WHERE
+    s.vet_id IS NULL
+    OR s.species_id IS NULL;
```

Comment:
- Maisy Smith doesn't have specialization so the number of visits should be more than 9 visits. Please 🙏🏽 fix this query to have the correct information.